var/spack/repos/builtin/packages/py-pythonsollya/package.py

```diff
@@ -23,3 +23,10 @@ class PyPythonsollya(PythonPackage):
     depends_on('sollya', type=('build', 'link'))
     depends_on('py-bigfloat', type=('build', 'run'))
     depends_on('mpfi', type=('build', 'link'))
+
+    @run_before('build')
+    def patch(self):
+        filter_file('PYTHON ?= python2',
+                    'PYTHON ?= python',
```

Comment:
```suggestion
                    'PYTHON ?= ' + self.spec['python'].command.path,
```
A `python` executable isn't always guaranteed to be present, Python 3 only installs `python3` by default.